Topics/jupyter-notebook.md

```diff
@@ -0,0 +1,154 @@
+
+This advanced (optional) tutorial will involve:
+
+- Warming up with the basics of shell commands and some handy magics, including a look at debugging, timing, and executing multiple languages.
+- Exploring topics like logging, macros, running external code, and Jupyter extensions.
+- Seeing how to enhance charts with [Seaborn](https://seaborn.pydata.org/), beautify notebooks with themes and CSS, and customise notebook output.
+- Finishing off with a deep look at topics like scripted execution, automated reporting pipelines, and working with databases.
+
+After these two tutorials you should be ready to use Jupyter Notebook in a pretty advanced way, leveraging all its useful capabilities.
+
+You can conclude your learning of Jupyter with this simple read:
+
+[**Ten simple rules for writing and sharing computational analyses in Jupyter Notebooks**](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007007)
+
+You can also find an [Example Repository](https://github.com/jupyter-guide/ten-rules-jupyter) with a coherent and clean Jupyter Notebooks examples. 
+
+See also: [Awesome Jupyter](https://github.com/markusschanta/awesome-jupyter) for a very detailed list of Jupyter projects, libraries and resources.
+
+#### JupyterLab
+
+When you use a Jupyter Notebook you just have the skeleton of an IDE, as it is designed for prototyping and analysis. 
+
+But what if you want a full IDE, fully integrated with Jupyter?
+
+_[JupyterLab](https://jupyterlab.readthedocs.io/en/stable/getting_started/overview.html) is the natural answer!_
+
+JupyterLab enables you to work with documents and activities such as Jupyter notebooks, text editors, terminals, and custom components in a flexible, integrated, and extensible manner.
+
+You can arrange multiple documents and activities side by side in the work area using tabs and splitters. Documents and activities integrate with each other, enabling new workflows for interactive computing.
+
+Here you have a [**deep tutorial about JupyterLab**](https://towardsdatascience.com/jupyter-lab-evolution-of-the-jupyter-notebook-5297cacde6b)!
+
+If you like [VSCode](https://code.visualstudio.com/), you can [use Jupyter Notebooks inside it!](https://www.google.com/search?q=vscode+and+jupyter&oq=vscode+and+jupyter+&aqs=chrome..69i57j0l7.2617j0j7&sourceid=chrome&ie=UTF-8)
+
+#### Colaboratory
+
+Remember that the Jupyter Notebook is nothing more than an application that runs on our machine, and therefore has its own computational capabilities.
+
+An alternative possibility is to use [**Google Colaboratory**](https://colab.research.google.com/notebooks/welcome.ipynb).
+
+Colaboratory is a free Jupyter notebook environment that requires no setup and runs entirely in the cloud. 
+
+With Colaboratory you can write and execute code, save and share your analyses, and access powerful computing resources, all for free from your browser. 
```

Comment:
You can also import and export notebooks and export python code from Colab.

Reply:
cool